### PR TITLE
Added only latest arrears action to list item

### PR DESCRIPTION
**Pull request by StevenLeighton21**
- Return only the most recent arrears action with list item
- Refactored tests slightly

### CODE REVIEW

File: LBHTenancyAPI/Domain/TenancyListItem.cs [hunk 1]
```diff
@@ -13,5 +13,60 @@ public struct TenancyListItem
         public string PrimaryContactName { get; set; }
         public string PrimaryContactShortAddress { get; set; }
         public string PrimaryContactPostcode { get; set; }
+
+        public TenancyListItem mergeWith(TenancyListItem t)
+        {
+            if (t.TenancyRef != null && TenancyRef == null)
```
Comment (VladyslavAtamanyuk) [inline LBHTenancyAPI/Domain/TenancyListItem.cs]: Just wanted to clarify, are we assuming that the TenancyListItem TenancyRef will always have a value and never be null?

Comment (LBHRShetty) [reply LBHTenancyAPI/Domain/TenancyListItem.cs]: cool.

Comment (StevenLeighton21) [reply LBHTenancyAPI/Domain/TenancyListItem.cs]: We're assuming self is truthy and that t will have some values to replace null values in self, so we only grab those where that is the case :)

File: LBHTenancyAPITest/Test/Gateways/UhTenanciesGatewayTest.cs [hunk 2]
```diff
@@ -83,20 +94,24 @@ private TenancyListItem CreateRandomTenancyListItem()
...
+            InsertTenancyAttributes(tenancy);
+
+            return tenancy;
+        }
+
         private void InsertTenancyAttributes(TenancyListItem tenancyAttributes)
         {
             string commandText =
-                "INSERT INTO araction (tag_ref, action_code, action_date) VALUES (@tenancyRef, @lastActionType, @lastActionTime);" +
+
```
Comment (VladyslavAtamanyuk) [inline LBHTenancyAPITest/Test/Gateways/UhTenanciesGatewayTest.cs]: Wild empty line

File: LBHTenancyAPI/Gateways/UhTenanciesGateway.cs [hunk 3]
```diff
@@ -19,27 +20,49 @@ public UhTenanciesGateway(string connectionString)
...
+
+                result.TenancyRef = row[0].TenancyRef;
+                result.LastActionCode = row[0].LastActionCode;
+                result.LastActionDate = row[0].LastActionDate;
+                result = tenancyListItem.mergeWith(result);
+
+                results.Add(result);
+            }
+
+            return results;
```
Comment (zuren) [inline LBHTenancyAPI/Gateways/UhTenanciesGateway.cs]: Why is this necessary? Feel like we ought to be able to:
 - Order by arag.start_date and araction.action_date to make the top result contain the latest of each.
- Make rows distinct on tag_ref to ensure we only get one result per tenancy.

File: LBHTenancyAPITest/Test/Gateways/UhTenanciesGatewayTest.cs [hunk 4]
```diff
@@ -57,6 +52,22 @@ public void WhenGivenTenancyRef_GetTenanciesByRefs_ShouldReturnTheLatestAgreemen
...
         }
 
+        [Fact]
+        public void WhenGivenTenancyRef_GetTenanciesByRefs_ShouldReturnTheLatestArrearsAction()
+        {
+            TenancyListItem expectedTenancy = InsertRandomisedTenancyListItem();
+
+            DateTime latestActionDate = expectedTenancy.LastActionDate.AddDays(1);
+            InsertArrearsActions(expectedTenancy.TenancyRef, "ABC",
+                expectedTenancy.LastActionDate.Subtract(DAY_IN_TIMESPAN));
```
Comment (zuren) [inline LBHTenancyAPITest/Test/Gateways/UhTenanciesGatewayTest.cs]: The `Subtract` isn't necessary, right?

File: LBHTenancyAPITest/Test/Gateways/UhTenanciesGatewayTest.cs [hunk 5]
```diff
@@ -125,5 +141,22 @@ private void InsertAgreement(string tenancyRef, string status, DateTime startDat
 
             command.ExecuteNonQuery();
         }
+
+
```
Comment (VladyslavAtamanyuk) [inline LBHTenancyAPITest/Test/Gateways/UhTenanciesGatewayTest.cs]: Wild empty line here

File: LBHTenancyAPITest/Test/Gateways/UhTenanciesGatewayTest.cs [hunk 6]
```diff
@@ -57,6 +52,22 @@ public void WhenGivenTenancyRef_GetTenanciesByRefs_ShouldReturnTheLatestAgreemen
...
+            TenancyListItem expectedTenancy = InsertRandomisedTenancyListItem();
+
+            DateTime latestActionDate = expectedTenancy.LastActionDate.AddDays(1);
+            InsertArrearsActions(expectedTenancy.TenancyRef, "ABC",
+                expectedTenancy.LastActionDate.Subtract(DAY_IN_TIMESPAN));
+            InsertArrearsActions(expectedTenancy.TenancyRef, "XYZ", latestActionDate);
+
+            var tenancies = GetTenanciesByRef(new List<string> {expectedTenancy.TenancyRef});
+
+            Assert.Equal(tenancies[0].LastActionDate, latestActionDate);
```
Comment (zuren) [inline LBHTenancyAPITest/Test/Gateways/UhTenanciesGatewayTest.cs]: Wonder if we should have a guarantee that this is using date to sort, instead of insertion order? At the moment, it could be either.

File: LBHTenancyAPITest/Test/Gateways/UhTenanciesGatewayTest.cs [hunk 7]
```diff
@@ -125,5 +141,22 @@ private void InsertAgreement(string tenancyRef, string status, DateTime startDat
 
             command.ExecuteNonQuery();
         }
+
```
Comment (VladyslavAtamanyuk) [inline LBHTenancyAPITest/Test/Gateways/UhTenanciesGatewayTest.cs]: Another Wild Line Here

File: LBHTenancyAPITest/Test/Gateways/UhTenanciesGatewayTest.cs (hunk 4 above)
Comment (StevenLeighton21) [reply LBHTenancyAPITest/Test/Gateways/UhTenanciesGatewayTest.cs]: Seems reasonable to have a randomised date then one entry before it and one after

File: LBHTenancyAPITest/Test/Gateways/UhTenanciesGatewayTest.cs (hunk 6 above)
Comment (StevenLeighton21) [reply LBHTenancyAPITest/Test/Gateways/UhTenanciesGatewayTest.cs]: No, it's adding the date we should get last, where the query unsorted would return the rows FIFO, no?
